File: nomad-samples/js/account-authenticaton/account/reset-password.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function resetPassword(USERNAME, TOKEN) 
{
		// Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${sessionStorage.getItem("token")}`);
  
  	const BODY = {
        userName: USERNAME,
        token: TOKEN,
        newPassword: "NewPassword123!"
    };
  
  // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/account/reset-password`, {
        method: "POST",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
      return(RESPONSE.status);
    }
		
  	// There was an error
      await apiExceptionHandler(RESPONSE, "Reset password failed");
}

File: nomad-samples/python/account_authentication/account/reset_password.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def reset_password(USENAME: str, TOKEN: str, NEW_PASSWORD: str) -> None:
    if not USENAME:
        raise Exception("USENAME: The username is invalid")
    
    if not TOKEN:
        raise Exception("Token: The token is invalid")
    
    if not NEW_PASSWORD:
        raise Exception("Password: The password is invalid")
  
    API_URL = PORTAL_API_URL + "/account/reset-password"
  
    # Create header for the request
    HEADERS = {
        "Content-Type": "application/json"
    }

    # Build the payload body
    BODY = {
        "userName": USENAME,
        "token": TOKEN,
        "newPassword": NEW_PASSWORD
    }
    
    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        
        if not RESPONSE.ok:
            raise Exception()
    except:
        raise Exception("Reset Password failed")